app.js

```diff
@@ -61,10 +59,34 @@ function fetchMeal(type){
         .then( res => {
             createMealCards(res.meals);           
             window.scrollTo(0,$('#mealCardsSection').offset().top);
+            $('#userInput').text($.trim($('#searchRecipe').val()));
+            setCache(res.meals, type);
         })
         .catch( e => console.warn(e) );
     }
-    $('#userInput').text($.trim($('#searchRecipe').val()));
+}
+
+//Function to save the data in the cache
+const setCache = (meal, type) => {
+    let mealJson = JSON.stringify(meal);
+    localStorage.setItem(type, mealJson);
```

Comment:
Sure, when could it be removed? I think it would be good to delete it when the tab is closed, but it would be the same as changing it for a sessionStorage. Or will there be a button to clear the search?

Reply:
If we use sessionStorage then everytime user reopens the tab, we are hitting the API unnecessarily even when we can have the data in cache. What about implementing native cache system and set it to expire in 3 days as default? Or can the same be done for localStorage variables?

Reply:
I'll use the cookies, with localStorage is complicated to set the expiration.

Reply:
Yeah go ahead. Set it to default expire in 3 days. What say?

Reply:
Yeah, 3 days is fine

Reply:
I check and the cookie cannot store more than 4kb, the search result is not saved for that reason, I will use the cookie to save only the random recipe with an expiration of 3 days and the sessionStorage for searches, this is deleted when close the browser. It's ok?

Reply:
Yes, go ahead :)